### PR TITLE
Add area rule

### PR DESCRIPTION
**Pull request by neilbmclaughlin**
Updated parcel schema to reflect changes in calculation engine.
Add rules/facts/tests for area.
Refatactor the for legibility

### CODE REVIEW

**Comment by neilbmclaughlin** [comment]
Fixed broken samples
Enforced perimeterTotal and perimeterFeatures in schema
Refactored tests for parcel creation

File: test/ffc-rules-engine.test.js [hunk 1]
```diff
@@ -176,42 +137,120 @@ describe('Rule: Claimed perimeter <= actual perimeter', () => {
...
+        const returnedFactsResults = ruleResult.conditions.all.map((c) => c.factResult)
+        expect(returnedFactsResults.length).toBe(1)
+        expect(returnedFactsResults[0]).toBe(60)
+      })
+      .run({ parcel, quantity: 0 })
+  })
+})
+
+describe('Rule: Claimed perimeter <= perimeter (within accepted tolerance)', () => {
+  const parcel = getParcelWithDefaults({})
```
Comment (djbingham) [inline test/ffc-rules-engine.test.js]: I think we should specify the parcel perimeter here to ensure that the perimeter with tolerance tests are valid regardless of the default parcel data.

File: test/ffc-rules-engine.test.js [hunk 2]
```diff
@@ -244,179 +271,159 @@ describe('Rule: Claimed perimeter <= perimeter adjusted to take into account per
...
-          type: 'lake',
-          perimeter: 15
-        }
-      ],
-      landCoverClass: 100,
-      previousActions: [],
-      sssi: false
-    }
+    const randomNonCulivatedClass = 100
+    const parcel = getParcelWithDefaults({ landCoverClass: randomNonCulivatedClass })
```
Comment (djbingham) [inline test/ffc-rules-engine.test.js]: Typo (missing `t`): `randomNonCulivatedClass` => `randomNonCultivatedClass`

File: test/ffc-rules-engine.test.js [hunk 3]
```diff
@@ -244,179 +271,159 @@ describe('Rule: Claimed perimeter <= perimeter adjusted to take into account per
...
-          perimeter: 15
-        }
-      ],
-      previousActions: [],
-      sssi: false
-    }
+})
+
+describe('Rule: Claimed area <= actual area', () => {
+  const parcel = getParcelWithDefaults({})
```
Comment (djbingham) [inline test/ffc-rules-engine.test.js]: I think we should specify the parcel area here to ensure that the area tests are valid regardless of the default data.

File: test/ffc-rules-engine.test.js [hunk 4]
```diff
@@ -244,179 +271,159 @@ describe('Rule: Claimed perimeter <= perimeter adjusted to take into account per
...
-    parcelRef: 'PR123',
-    perimeter: 75,
-    perimeterFeatures: [],
-    previousActions: [],
-    sssi: false
-  }
-  test('Passes when claimed perimeter is less than actual perimeter (allowing for tolerance)', async () => {
+describe('Rule: Claimed area <= area adjusted to take into account area features', () => {
+  test('Passes when there are no area features and claimed area less than area', async () => {
+    const parcel = getParcelWithDefaults({})
```
Comment (djbingham) [inline test/ffc-rules-engine.test.js]: Again, I think we should specify parcel area here.

File: test/ffc-rules-engine.test.js [hunk 5]
```diff
@@ -244,179 +271,159 @@ describe('Rule: Claimed perimeter <= perimeter adjusted to take into account per
...
     )
 
     expect(result.events.length).toBe(1)
-    expect(result.events[0].type).toBe('withinTolerancePerimeter')
+    expect(result.events[0].type).toBe('withinPondlessArea')
   })
-  test('Passes when claimed perimeter is equal to actual perimeter (allowing for tolerance)', async () => {
+  test('Passes when claimed area is equal to adjusted area', async () => {
+    const parcel = getParcelWithDefaults({
+      areaFeatures: [
```
Comment (djbingham) [inline test/ffc-rules-engine.test.js]: And specify total parcel area here.

File: test/ffc-rules-engine.test.js [hunk 6]
```diff
@@ -244,179 +271,159 @@ describe('Rule: Claimed perimeter <= perimeter adjusted to take into account per
...
     )
 
     expect(result.events.length).toBe(1)
-    expect(result.events[0].type).toBe('withinTolerancePerimeter')
+    expect(result.events[0].type).toBe('withinPondlessArea')
   })
-  test('Fails when claimed perimeter is greater than actual perimeter (allowing for tolerance)', async () => {
+  test('Fails when claimed area is greater than adjusted area', async () => {
+    const parcel = getParcelWithDefaults({
+      areaFeatures: [
```
Comment (djbingham) [inline test/ffc-rules-engine.test.js]: And specify total parcel area here.

**Pull request merged**